### PR TITLE
Switching to using last time and current time for velocity calculation

### PR DESCRIPTION
**Pull request by preston-rogers**
Added changes to the platinum and gold drive offline modes to have more accurate velocity calculation.
Before this MR, velocity was calculated using the loop period which may be inaccurate due to jitter. Now the velocity is calculated using the prior process loop time and the current process loop time.

### CODE REVIEW

File: src/device_base.cc [hunk 1]
```diff
@@ -27,6 +27,7 @@ void fastcat::DeviceBase::SetLoopPeriod(double loop_period)
 
 void fastcat::DeviceBase::SetTime(double time, double monotonic_time)
 {
+  last_time_             = state_->monotonic_time;
```
Comment (kwehage) [inline src/device_base.cc:30]: you need to differentiate between monotonic and standard time, they are two different clock sources..

can you change this variable to `last_monotonic_time_` ?

File: src/fastcat_devices/three_node_thermal_model.cc [hunk 2]
```diff
@@ -169,7 +169,6 @@ FaultType ThreeNodeThermalModel::Process()
   state_->three_node_thermal_model_state.node_2_temp = node_temps_[1];
   state_->three_node_thermal_model_state.node_3_temp = node_temps_[2];
   state_->three_node_thermal_model_state.node_4_temp = node_temps_[3];
-  last_time_ = state_->time;  // update loop time
```
Comment (kwehage) [inline src/fastcat_devices/three_node_thermal_model.cc:172]: You are mixing up standard time and monotonic time here. state_->time is not using monotonic_time_

**Review by kwehage — COMMENTED** [review]
See comments

File: src/fastcat_devices/three_node_thermal_model.cc (hunk 2 above)
Comment (preston-rogers) [reply src/fastcat_devices/three_node_thermal_model.cc:172]: Just made the switches. Thanks for catching this :)

File: src/device_base.cc (hunk 1 above)
Comment (preston-rogers) [reply src/device_base.cc]: Done!

**Pull request merged**